### PR TITLE
[FR] Simple API endpoint to change own user metadata

### LINKED ISSUE #3979: [FR] :checkered_flag: Simple API endpoint to change user metadata
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### Problem statement

we currently provide an API endpoint for editing user data via a form. Fetching and changing endpoints are different and take different parameters.

### Suggested solution

I think it would be better to provide a RetrieveUpdateAPI endpoint under `/api/user/me` (somewhat a convention). This endpoint could provide and patch all the metadata for a user.

### Describe alternatives you've considered

N/A

### Examples of other systems

_No response_

### Do you want to develop this?

- [X] I want to develop this.

### PR DESCRIPTION
**Pull request by matmair**
Fixes #3979

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/4151"></a>

### CODE REVIEW

**Comment by matmair** [comment]
@SchrodingersGat experiencing some ... interesting test behaviour here - any idea why a test fails if all tests are run but succeeds if only the specific file is run (only on pgsql)?

**Comment by SchrodingersGat** [comment]
Nope, that's a new one for me. Something being set in the test environment by a prior test?

**Comment by matmair** [comment]
Maybe, not really sure. Let's see if it now works.

File: InvenTree/users/tests.py [hunk 1]
```diff
@@ -242,3 +259,7 @@ def test_token(self):
         # token second delete
         response = self.client.delete(reverse('api-token'), {}, format='json')
         self.assertEqual(response.status_code, 400)
+
+        # test user is associated with token
+        respone_me = self.do_request(reverse('api-user-me'), {}, 200)
```
Comment (SchrodingersGat) [inline InvenTree/users/tests.py:264]: spelling "response"

**Comment by SchrodingersGat** [comment]
Have you worked out the CI issues here?

**Comment by matmair** [comment]
@SchrodingersGat no, now I know why it was commented out originally

**Comment by SchrodingersGat** [comment]
@matmair is there any intent in this PR to replace the front-end forms with this API driven approach? Or would you like to see this merged separately "as is"

**Comment by matmair** [comment]
@SchrodingersGat no, this is more to get in line with the expectations OIDC / OAuth2 connector have regarding user info endpoints

**Pull request merged**